paymentsheet/src/main/java/com/stripe/android/paymentsheet/ExternalPaymentMethodResultHandler.kt

```diff
@@ -0,0 +1,126 @@
+package com.stripe.android.paymentsheet
+
+import android.content.Context
+import android.content.Intent
+import android.os.Parcelable
+import androidx.annotation.RestrictTo
+import com.stripe.android.core.exception.LocalStripeException
+import com.stripe.android.payments.core.analytics.ErrorReporter
+import com.stripe.android.payments.paymentlauncher.PaymentResult
+import kotlinx.parcelize.Parcelize
+import java.lang.IllegalStateException
+
+/**
+ * Handler used to respond to external payment method confirm results.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+object ExternalPaymentMethodResultHandler {
+
+    /**
+     * Updates the PaymentSheet UI to reflect the result of confirming an external payment method.
+     *
+     * Should be called when [ExternalPaymentMethodConfirmHandler.confirmExternalPaymentMethod] completes.
+     */
+    fun onExternalPaymentMethodResult(context: Context, externalPaymentMethodResult: ExternalPaymentMethodResult) {
+        fun updatePaymentSheetWithResult() {
```

Comment:
What if the merchant calls this and payment sheet isn't in the backstack? Can you write a tests that we just ignore it in the activity?

Reply:
This should be solved now that we are using an activity to handle calling the external payment method confirm handler and listening for the result. 

If someone were to call this function now without PaymentSheet in the backend, the ExternalPaymentMethodActivity would start and immediately finish with a result. Nothing would be listening for that result, so nothing would happen.